Tasks/Xcode/xcode.ts

```diff
@@ -178,14 +179,27 @@ async function run() {
         else if (signingOption === 'manual') {
             xcode_codeSignStyle = 'CODE_SIGN_STYLE=Manual';
 
-            let signIdentity: string = tl.getInput('signingIdentity');
+            const signIdentity: string = tl.getInput('signingIdentity');
             if (signIdentity) {
                 xcode_codeSignIdentity = 'CODE_SIGN_IDENTITY=' + signIdentity;
             }
 
             let provProfileUUID: string = tl.getInput('provisioningProfileUuid');
-            if (provProfileUUID) {
-                xcode_provProfile = 'PROVISIONING_PROFILE=' + provProfileUUID;
+            let provProfileName: string = tl.getInput('provisioningProfileName');
+
+            // If either value is set, override both .pbxproj settings. Testing indicates that
+            // PROVISIONING_PROFILE_SPECIFIER take precedence, if it's not empty.
+            if (provProfileUUID || provProfileName) {
```

Comment:
Patterns like this have generally caused problems. It is better to always pass whatever value is specified or have an explicit way for user to specify which one to use. Can we remove the if statement?